os/kernel/init/os_start.c

```diff
@@ -194,31 +194,23 @@ volatile dq_queue_t g_waitingforsemaphore;
 
 /* This is the list of all tasks that are blocked waiting for a signal */
 
-#ifndef CONFIG_DISABLE_SIGNALS
 volatile dq_queue_t g_waitingforsignal;
```

Comment:
To support enabled CONFIG_DISABLE_SIGNALS, you define the queue always regardless of the config.
But when the config is enabled, this will not be used.

When the config is enabled, if we give null at the g_tasklisttable, what happen?
I mean "is there any problem on usages and memory?"

Reply:
Since we are deleting the config check in tstate enum, if we add null in this array, then it will cause problem.

Reply:
When CONFIG_DISABLE_SIGNALS is enabled and enum has no config checking, scheduler will not access the queue. Because the config is enabled (the functionality is disabled).
But if we don't match the number of enum and queue list, normal operation can have mismatched.
To resolve it, we have two options.
1. this PR - just enable all lists
2. my suggestion - keep config checking at the queue list, but add null if the config is enabled.

I told you 2nd option. I think the scheduler will not access the list (null) when the config is enabled (function is disabled). It means no issue with 2nd option. Could you correct me something wrong in my opinion?